Solutions/Corvus.Tenancy.Storage.Azure.Blob/Corvus/Tenancy/Storage/Azure/TenantProviderBlobStore.cs

```diff
@@ -189,6 +203,19 @@ public async Task<ITenant> CreateChildTenantAsync(string parentTenantId, string
             {
                 throw new TenantNotFoundException();
             }
+            catch (StorageException ex) when (ex.RequestInformation.HttpStatusCode == (int)HttpStatusCode.Conflict)
+            {
+                // This exception is thrown because there's already a tenant with the same Id. This should never happen; when
+                // this method has been called from CreateChildTenantAsync, then the Guid will have been generated and the
+                // chances of it matching one previously generated are miniscule. However, it could happen when calling this
+                // method directly with a wellKnownChildTenantGuid that's already in use. In this case, the fault is with
+                // the client code - creating tenants with well known Ids is something one would expect to happen under
+                // controlled conditions, so it's only likely that a conflict will occur when either the client code has made
+                // a mistake or someone is actively trying to cause problems.
+                throw new ArgumentException(
+                    $"A child tenant of '{parentTenantId}' with a well known Guid of '{wellKnownChildTenantGuid}' already exists.",
```

Comment:
Those are globally unique IDs, but not GUIDs in the RFC sense of a GUID, so the error should say "ID".

Reply:
In this case, we're talking about the GUID that's been passed into the function, rather than the resulting child tenant Id, so makes sense to leave as is.